pkg/controller/cloud/node_controller.go

```diff
@@ -321,7 +321,7 @@ func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
 		}
 
 		if zones, ok := cnc.cloud.Zones(); ok {
-			zone, err := zones.GetZone()
+			zone, err := getZoneByProviderIDOrName(zones, curNode)
```

Comment:
+1
Do we update (https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L310) too?